### PR TITLE
[PLGN-786] Netskope - Updated SDK to the latest version | Fixed issue related to API changes | Updated unittests

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Updated SDK to the latest version
  - Fixed issue related to API changes
  - Updated unittests

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Testing

Every action was tested locally

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Comment by igorski-r7** [comment]
The `PluginValidator` is failing due to the change of `type` but it got updated from `object` to `data` which allows choosing the output from UI perspective. It's not breaking change so I think we could leave version `1.0.2` as it is instead of bumping it to `2.0.0`.

**Review by jerichardson-r7 — APPROVED** [review]

File: plugins/netskope/plugin.spec.yaml [hunk 1]
```diff
@@ -21,6 +22,32 @@ hub_tags:
   use_cases: [cloud_security]
   keywords: [endpoint]
   features: []
+sdk:
+  type: slim
+  version: 5.4.5
+  user: nobody
+key_features:
+  - Apple currently pending changes for URL lists
```
Comment (cmcnally-r7) [inline plugins/netskope/plugin.spec.yaml:30]:
```suggestion
  - Apply currently pending changes for URL lists
```

File: plugins/netskope/icon_netskope/actions/replace_url_list_by_id/action.py [hunk 2]
```diff
@@ -13,8 +13,26 @@ def __init__(self):
...
+            },
+        )
+        return {
+            Output.ID: response.get("id"),
+            Output.NAME: response.get("name"),
+            Output.DATA: response.get("data"),
+            Output.PENDING: response.get("pending"),
+            Output.MODIFY_BY: response.get("modify_by"),
+            Output.MODIFY_TIME: response.get("modify_time"),
+            Output.MODIFY_TYPE: response.get("modify_type"),
```
Comment (cmcnally-r7) [inline plugins/netskope/icon_netskope/actions/replace_url_list_by_id/action.py:37]: Is it worth adding default values here
```suggestion
            Output.ID: response.get("id"),
            Output.NAME: response.get("name"),
            Output.DATA: response.get("data"),
            Output.PENDING: response.get("pending"),
            Output.MODIFY_BY: response.get("modify_by"),
            Output.MODIFY_TIME: response.get("modify_time"),
            Output.MODIFY_TYPE: response.get("modify_type", "") etc etc,
```

File: plugins/netskope/icon_netskope/actions/patch_url_list_by_id/action.py [hunk 3]
```diff
@@ -13,7 +13,24 @@ def __init__(self):
...
+            data["data"] = {"urls": urls, "type": type_}
+        response = self.connection.client.patch_url_list_by_id(identifier, action, data)
+        return {
+            Output.ID: response.get("id"),
+            Output.NAME: response.get("name"),
+            Output.DATA: response.get("data"),
+            Output.PENDING: response.get("pending"),
+            Output.MODIFY_BY: response.get("modify_by"),
+            Output.MODIFY_TIME: response.get("modify_time"),
+            Output.MODIFY_TYPE: response.get("modify_type"),
```
Comment (cmcnally-r7) [inline plugins/netskope/icon_netskope/actions/patch_url_list_by_id/action.py:35]: same here

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**